lib/runner.coffee

```diff
@@ -95,21 +101,24 @@ class Runner
       arg = arg.replace(/{FILE_ACTIVE_NAME_BASE}/g, path.join(codeContext.filename, '..'))
     if project_path?
       arg = arg.replace(/{PROJECT_PATH}/g, project_path)
-    
+
     arg
 
   args: (codeContext, extraArgs) ->
     args = (@scriptOptions.cmdArgs.concat extraArgs).concat @scriptOptions.scriptArgs
-    project_path = ''
-    paths = atom.project.getPaths()
-    if paths.length > 0
-      fs.stat(paths[0], (err, stats) ->
-        if !err
-          project_path = if stats.isDirectory() then paths[0] else path.join(paths[0], '..')
-      )
-    
+    project_path = @getProjectPath or ''
     args = (@fillVarsInArg arg, codeContext, project_path for arg in args)
-    
+
     if not @scriptOptions.cmd? or @scriptOptions.cmd is ''
       args = codeContext.shebangCommandArgs().concat args
     args
+
+  getProjectPath: ->
```

Comment:
Because we think about overall perfomance and speed ;)

This version of function work ok for me
```coffee
  getProjectPath: ->
    filePath = atom.workspace.getActiveTextEditor().getPath()
    project_path = ''
    projectPaths = atom.project.getPaths()
    if projectPaths?.length > 0
      fs.stat(projectPaths[0], (err, stats) ->
        if !err
          project_path = if stats.isDirectory() then projectPaths[0] else path.join(projectPaths[0], '..')
      )
    project_path
```

**but** i can't understand your logic. you make loop, take the first element, check its' index... but it always `> -1` and you always get the first element.